modules/parachains/src/call_ext.rs

```diff
@@ -133,24 +243,23 @@ pub trait CallSubType<T: Config<I, RuntimeCall = Self>, I: 'static>:
 	/// block production, or "eat" significant portion of block production time literally
 	/// for nothing. In addition, the single-parachain-head-per-transaction is how the
 	/// pallet will be used in our environment.
-	fn check_obsolete_submit_parachain_heads(&self) -> TransactionValidity
+	fn check_obsolete_submit_parachain_heads(
```

Comment:
Nit: The same as for the grandpa call ext. I think I would rename this to something like `check_executable_submit_parachain_heads` and move the logic in `SubmitParachainHeadsHelper::check_obsolete_from_extension` here. Also would rename `VerifiedSubmitParachainHeadsInfo` -> `ExecutableSubmitParachainHeadsInfo`.

Reply:
Same as here: https://github.com/paritytech/parity-bridges-common/pull/2873#discussion_r1560909073